dali/pipeline/operator/builtin/external_source.h

```diff
@@ -450,10 +444,10 @@ class ExternalSource : public Operator<Backend>, virtual public BatchSizeProvide
   inline std::enable_if_t<std::is_same<B, CPUBackend>::value>
   CopyUserData(const SourceDataType<SrcBackend> &batch,
```

Comment:
A general remark: we can get rid of most, if not all, usages of `template<typename> class SourceDataType` and possibly inline those "helpers" which seem redundant now.

Reply:
Removed it form [Copy/Share]UserData, ValidateInputData, SetDataSourceHelper.